src/rulesets/SparseArrays/array.jl

```diff
@@ -0,0 +1,16 @@
+using SparseArrays
+
+function rrule(::Type{<:SparseMatrixCSC{T,N}}, arr) where {T,N}
+  function SparseMatrix_pullback(Δ)
+    (collect(Δ),)
+  end
+  SparseMatrixCSC{T,N}(arr), SparseMatrix_pullback
+end
+
+function rrlue(::typeof(diagm), x)
+  diagm(x), d -> (diag(d),)
+end
+
+function rrlue(::typeof(issymmetric), x)
+  issymmetric(x), d -> (d,)
+end
```

Comment:
similar comments to above: please move to LinearAlgebra, add type constraints and follow the style guide :)